app/uk/gov/hmrc/gform/sharedmodel/TaxPeriods.scala

```diff
@@ -46,14 +45,14 @@ object ObligationDetail {
   implicit val format: OFormat[ObligationDetail] = Json.format[ObligationDetail]
 }
 
-case class ObligationDetails(obligationDetails: NonEmptyList[ObligationDetail])
+case class ObligationDetails(obligationDetails: List[ObligationDetail])
 
 object ObligationDetails {
   import JsonUtils._
   implicit val format: OFormat[ObligationDetails] = Json.format[ObligationDetails]
 }
 
-case class Obligation(obligations: NonEmptyList[ObligationDetails])
+case class Obligation(obligations: List[ObligationDetails])
```

Comment:
Why has this been changed to `List`?

Reply:
The reason this is a list is that It needs to be the same as the backend and on the backend it is also a list.
The reason it's a list on the backend is that when the api is called and the periods are retrieved they are stored in this list. In the case there are no periods then the list is empty.
If there's another way to deal with retrieving no tax periods then I'm not really sure what it is

Reply:
This is actually the reason I can't change what the other two are as this being a List and the others being NonEmptyList doesn't seem to work (or at least I can't get them to)